# Intern Name : Ashutosh Bafna

The Sparks Foundation - Data Science & Business Analytics Internship

Task : Prediction using decision tree algorithm 

Problem statement : if any new data is feed it will be able to predict class accordingly 

In [73]:
import pandas as pd
import numpy as np

In [74]:
iris = pd.read_csv(r"D:\SPARK INTERNSHIP\task_dt\Iris.csv")  #dataset 

In [75]:
iris.head()

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

In [76]:
iris.Species.unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

In [77]:
iris.isna().sum() #check is their any null value 

Id               0
SepalLengthCm    0
SepalWidthCm     0
PetalLengthCm    0
PetalWidthCm     0
Species          0
dtype: int64

In [78]:
#no_null values 

In [79]:
iris = iris.drop(['Id'],axis=1) #drop id column 

In [80]:
iris.head()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0            5.1           3.5            1.4           0.2  Iris-setosa
1            4.9           3.0            1.4           0.2  Iris-setosa
2            4.7           3.2            1.3           0.2  Iris-setosa
3            4.6           3.1            1.5           0.2  Iris-setosa
4            5.0           3.6            1.4           0.2  Iris-setosa

In [81]:
from sklearn.model_selection import train_test_split # Splitting the dataset into training and test set.  

In [82]:
iris_train, iris_test = train_test_split(iris, test_size=0.25) 

In [83]:
iris_train_x = iris_train.iloc[:,0:-1]
iris_train_y = iris_train.iloc[:,-1]

In [84]:
iris_test_x = iris_test.iloc[:,0:-1]
iris_test_y = iris_test.iloc[:,-1]

In [85]:
#spliting done 

In [86]:
from sklearn.tree import DecisionTreeClassifier 
dt = DecisionTreeClassifier() 

In [87]:
#Fitting Decision Tree classifier to the training set  

In [88]:
dt.fit(iris_train_x, iris_train_y)

DecisionTreeClassifier()

In [89]:
#Predicting the test set result  

In [90]:
pred_dt = dt.predict(iris_test_x)

# model evalution 

In [91]:
from sklearn.metrics import confusion_matrix

In [92]:
tab_dt = confusion_matrix(iris_test_y,pred_dt)
tab_dt

array([[11,  0,  0],
       [ 0, 15,  1],
       [ 0,  2,  9]], dtype=int64)

In [93]:
from sklearn.metrics import accuracy_score, precision_score

In [94]:
accuracy =accuracy_score(iris_test_y, pred_dt)*100
precision = precision_score(iris_test_y, pred_dt,average='weighted')*100 

In [95]:
from sklearn.metrics import recall_score, f1_score, roc_auc_score,roc_curve

In [96]:
recall=recall_score(iris_test_y, pred_dt,average='weighted')

f1=f1_score(iris_test_y, pred_dt,average='weighted') 

In [97]:
print("accuracy of model :" ,accuracy )
print("precision of model :", precision)
print("recall_score of model :", recall)
print("f1_score of model :" ,f1)

accuracy of model : 92.10526315789474
precision of model : 92.15170278637771
recall_score of model : 0.9210526315789473
f1_score of model : 0.920369104579631


Conclusion : Model is build and tested ; Accuracy and precision of model around 92% 

Thank you 